### PR TITLE
Add multi-arch image via Bazel

### PR DESCRIPTION
**Pull request by mattmoor**
**Description**

This starts to add multi-arch container images built via Bazel, turned into "manifest lists" in the Docker daemon, and published.

**Submitter Checklist**

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [unit tests](../DEVELOPMENT.md#creating-a-pr)
- [x] Adds integration tests if needed.

_See [the contribution guide](../CONTRIBUTING.md) for more details._

**Reviewer Notes**

- [ ] The code flow looks good. 
- [ ] Unit tests and or integration tests added.

**Release Notes**

```release-notes
The executor:bazel image now supports both amd64 and arm64.
```

### CODE REVIEW

**Comment by mattmoor** [comment]
I'm staging this to test the Travis integration, but I need to make the `cloudbuild.yaml` changes non-invasive (right now they are breaking, and should not be merged)

cc @tejal29

**Comment by mattmoor** [comment]
Travis seems to have not gotten notified, or is otherwise upset with me... 🤔

**Comment by mattmoor** [comment]
Alright, commenting the travis changes seems to have unwedged it.  I "love" silent failures 🙃

**Comment by tejal29** [comment]
Looks like we need to switch to paid tier for travis.

**Comment by mattmoor** [comment]
Alright, my stripped down Travis build passed, so I uncommented everything here: https://travis-ci.org/github/GoogleContainerTools/kaniko/builds/734073723

This should be RFAL, please look carefully at the `deploy/cloudbuild_release.yaml` as I did not test that due to hard-coded project and use of tags.

**Comment by mattmoor** [comment]
Everything is green! 🤩

File: cmd/executor/BUILD [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+load("@io_bazel_rules_docker//contrib:test.bzl", "container_test")
+
+# Non-executable tests can run on any architecture,
+# so do not tag them.
+[
+    container_test(
+        name = "image_files_" + arch + "_test",
+        configs = ["testdata/files.yaml"],
+        image = ":image_" + arch,
+        # tags = ["manual", arch],
```
Comment (tejal29) [inline cmd/executor/BUILD:84]: can we remove the comment?

File: cmd/executor/testdata/Dockerfile.trivial [hunk 2]
```diff
@@ -0,0 +1,2 @@
+FROM ubuntu
```
Comment (tejal29) [inline cmd/executor/testdata/Dockerfile.trivial:1]: this is a little bit odd. all our tests and test data is in 
integration/
Would it possible to move these file in integration ?

File: cmd/executor/BUILD [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    )
+    for arch in ARCHITECTURES
+]
+
+load("@io_bazel_rules_docker//contrib:test.bzl", "container_test")
+
+# Non-executable tests can run on any architecture,
+# so do not tag them.
+[
+    container_test(
```
Comment (tejal29) [inline cmd/executor/BUILD:80]: Can these move to integration package.

**Review by tejal29 — COMMENTED** [review]
LGTM except some nits on code organization.

File: cmd/executor/BUILD (hunk 1 above)
Comment (mattmoor) [reply cmd/executor/BUILD]: Done

File: cmd/executor/testdata/Dockerfile.trivial (hunk 2 above)
Comment (mattmoor) [reply cmd/executor/testdata/Dockerfile.trivial]: Done

File: cmd/executor/BUILD (hunk 3 above)
Comment (mattmoor) [reply cmd/executor/BUILD]: Done

**Comment by mattmoor** [comment]
Alright, looks like the previous failure was a flake, so this should be RFAL.

**Comment by mattmoor** [comment]
@tejal29 LMK if these needs anything else!

**Review by tejal29 — APPROVED** [review]

**Comment by tejal29** [comment]
I will share the results of GCB tirgger with you.

**Pull request merged**